examples/usecases/llm_diffusion_serving_app/docker/sd/stable_diffusion_handler.py

```diff
@@ -0,0 +1,183 @@
+            )
+
+        if compile_unet:
+            logger.info("Compiling UNet.")
+            if compile_mode == "max-autotune" and change_comp_config:
+                pipe.unet.to(memory_format=torch.channels_last)
+                torch._inductor.config.conv_1x1_as_mm = True
+                torch._inductor.config.coordinate_descent_tuning = True
+                torch._inductor.config.epilogue_fusion = False
+                torch._inductor.config.coordinate_descent_check_all_directions = True
+
+            pipe.unet = torch.compile(pipe.unet, **compile_options)
+
+        if compile_vae:
+            logger.info("Compiling VAE.")
+            if compile_mode == "max-autotune" and change_comp_config:
+                pipe.vae.to(memory_format=torch.channels_last)
+                torch._inductor.config.conv_1x1_as_mm = True
+                torch._inductor.config.coordinate_descent_tuning = True
+                torch._inductor.config.epilogue_fusion = False
+                torch._inductor.config.coordinate_descent_check_all_directions = True
+
+            pipe.vae.decode = torch.compile(pipe.vae.decode, **compile_options)
+
+        logger.info(f"Compiled {ckpt} model with {compile_options}")
+        pipe.set_progress_bar_config(disable=True)
+
+        self.pipeline = pipe
+        logger.info(f"Stable Diffusion model loaded successfully: {ckpt}")
+        self.initialized = True
+
+        return pipe
+
+    @timed
+    def preprocess(self, requests):
+        """Basic text preprocessing, of the user's prompt.
+        Args:
+            requests (str): The Input data in the form of text is passed on to the preprocess
+            function.
+        Returns:
+            list : The preprocess function returns a list of prompts.
+        """
+
+        assert (
+            len(requests) == 1
+        ), "Stable Diffusion currently only supported with batch_size=1"
+
+        req_data = requests[0]
+        input_data = req_data.get("data") or req_data.get("body")
+
+        if isinstance(input_data, (bytes, bytearray)):
+            input_data = input_data.decode("utf-8")
+
+        if isinstance(input_data, str):
+            input_data = json.loads(input_data)
+
+        return input_data
+
+    @timed
+    def inference(self, model_inputs):
+        """Generates the image relevant to the received text.
+        Args:
+            input_batch (list): List of Text from the pre-process function is passed here
+        Returns:
+            list : It returns a list of the generate images for the input text
+        """
+        # Handling inference for sequence_classification.
+        guidance_scale = model_inputs.get("guidance_scale") or 5.0
+        num_inference_steps = model_inputs.get("num_inference_steps") or 5
+        height = model_inputs.get("height") or 768
+        width = model_inputs.get("width") or 768
+        inferences = self.pipeline(
```

Comment:
Does the diffusion pipeline set torch.no_grad() ?

Reply:
Yes, it does.
https://huggingface.co/docs/diffusers/main/en/api/pipelines/overview

Reply:
Something that I have observed is that for compile, torch.inference_mode() gives better perf compared to torch.no_grad(). We don't need to address this in this PR

Reply:
Thanks for letting me know. Yes, I'm observing the same when I try to profile, it shows a different performance ! As with the profiler, it uses torch.inference_mode() !!